### PR TITLE
Fix reset causes an unhandled error

### PR DESCRIPTION
**Pull request by 5c077m4n**

### CODE REVIEW

**Comment by roee-shapira-tikal** [comment]
https://github.com/mosaic-org/mosaic/issues/43

**Comment by roee-shapira-tikal** [comment]
@imsnif ready :)

File: src/terminal_pane/terminal_pane.rs [hunk 1]
```diff
@@ -456,12 +456,7 @@ impl vte::Perform for TerminalPane {
             };
             self.scroll.move_cursor_back(move_back_count);
         } else if c == 'l' {
-            let first_intermediate_is_questionmark = match _intermediates.get(0) {
-                Some(b'?') => true,
-                None => false,
-                _ => false,
-            };
-            if first_intermediate_is_questionmark {
+            if _intermediates.get(0) == Some(&b'?') {
```
Comment (imsnif) [inline src/terminal_pane/terminal_pane.rs:459]: I think if we're using this we can remove the leading `_`, right?

File: src/terminal_pane/terminal_pane.rs [hunk 2]
```diff
@@ -471,15 +466,12 @@ impl vte::Perform for TerminalPane {
...
+                self.scroll.clear_all();
             }
         } else if c == 'h' {
-            let first_intermediate_is_questionmark = match _intermediates.get(0) {
-                Some(b'?') => true,
-                None => false,
-                _ => false,
-            };
-            if first_intermediate_is_questionmark {
+            if _intermediates.get(0) == Some(&b'?') {
```
Comment (imsnif) [inline src/terminal_pane/terminal_pane.rs:474]: Also - thanks for this. This is of course much better. I should not be coding in my sleep ;P

File: src/terminal_pane/terminal_pane.rs [hunk 3]
```diff
@@ -584,6 +576,19 @@ impl vte::Perform for TerminalPane {
...
+             */
+            if let Some(move_cursor_back) = params.get(1) {
+                let move_cursor_back = *move_cursor_back;
+                if move_cursor_back > 0 {
+                    self.move_cursor_backwards(move_cursor_back as usize);
+                }
+            }
+        } else if c == 'p' {
+            self.clear_scroll();
+            self.reset_size_and_position_override();
```
Comment (imsnif) [inline src/terminal_pane/terminal_pane.rs:591]: I'm not sure I understand... we use `reset_size_and_position_override` when the pane was full_screen (had a size and position override) and we want to bring it back. Why do we need it in this case?

**Review by imsnif — CHANGES_REQUESTED** [review]
Looking good. Left a few small comments.
Also: how about writing a test for this?
You can check out the compatibility tests if you like. Essentially what they do is run a fixture (which we get from running mosaic with the `--debug` flag) and comparing it against a snapshot to make sure it is as expected. Want to try and write one? I can provide a more detailed explanation if you like.

File: src/terminal_pane/terminal_pane.rs (hunk 3 above)
Comment (roee-shapira-tikal) [reply src/terminal_pane/terminal_pane.rs:591]: Sorry, I saw `reset` and thought: reset!

**Comment by roee-shapira-tikal** [comment]
@imsnif I'll try on my own to add a test and when I'll need help I'll tell you

**Comment by imsnif** [comment]
This has been already been fixed.

**Pull request closed without merging**